### PR TITLE
mimic: rgw: when using nfs-ganesha to upload file, rgw es sync module get failed

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/37348

### CODE REVIEW

File: src/rgw/rgw_sync_module_es.cc [hunk 1]
```diff
@@ -304,19 +319,30 @@ struct es_obj_metadata {
             }
           }
         }
-      } else if (name == "x-amz-tagging") {
-        auto tags_bl = val.begin();
-        decode(obj_tags, tags_bl);
-      } else if (name == "compression") {
+      } else if (attr_name == RGW_ATTR_TAGS) {
+        try {
+          auto tags_bl = val.begin();
```
Comment (pdvian) [inline src/rgw/rgw_sync_module_es.cc]: @theanalyst Need review. the bufferlist cbegin() is not available in mimic.

**Comment by smithfarm** [comment]
@theanalyst Can you help @pdvian here?

@pdvian Maybe refer to https://github.com/ceph/ceph/pull/25444 (luminous backport of this same fix)?

**Comment by theanalyst** [comment]
Can you backport https://github.com/ceph/ceph/pull/25161 as well onto this so that clang doesnt complain

**Comment by theanalyst** [comment]
Alternatively just adding a <const char*> on the initializer_list should work

**Comment by smithfarm** [comment]
@pdvian Could you respond to @theanalyst 's comments from December 13, please?

File: src/rgw/rgw_sync_module_es.cc (hunk 1 above)
Comment (theanalyst) [reply src/rgw/rgw_sync_module_es.cc]: This is ok

Comment (theanalyst) [reply src/rgw/rgw_sync_module_es.cc]: Please see comments on const char* for initializer_list

**Comment by pdvian** [comment]
@smithfarm @theanalyst Let me do the requested changes.

**Comment by smithfarm** [comment]
@pdvian Please rebase

**Comment by smithfarm** [comment]
closing in favor of https://github.com/ceph/ceph/pull/27972

**Pull request closed without merging**